thumbor/media.py

```diff
@@ -51,35 +88,53 @@ def file_extension(self):
 
     @property
     def mime(self):
-        mime = None
+        if self._mimetype:
```

Comment:
`mime` is not called often and any changes to the `buffer` property has a chance of altering the value. It's probably best to not cache into `_mimetype`.